### ISSUE TITLE
Parser hangs on stream of `%Y`

### ISSUE THREAD

**Issue by ingydotnet** [aaubry/YamlDotNet#638]
Found this bug while trying things in <https://spec.yaml.io/main/playground/parser>.

It froze the entire page because of it. I'll need to add a timeout certainly but thought you should know.

**To Reproduce**
```
docker run --rm -it yamlio/yaml-editor bash -c 'echo -n "%Y" | /yaml/bin/dotnet-yamldotnet-event'
```

**Comment by dmitry-ra** [comment]
Bug was planted in commit https://github.com/aaubry/YamlDotNet/pull/420/commits/08c929f3c3c2d2cacdd12fa01269c6eb49fa2ca9 when reserved directive support was added.

This `while` loop is endless - analyzer uses buffer, that don't care about overflows and ranges.
```c#
while (!analyzer.Check('#') && !analyzer.IsBreak())
{
    Skip();
}
```

**Comment by ingydotnet** [comment]
You can see it here: https://play.yaml.io/main/parser?input=JVk=

Playing around in there it seems to hang on directive line (including reserved) that does not have a line break.

See https://play.yaml.io/main/parser?input=I3h4eApmb286IGJhcgouLi4KJXh5eiBAJCVeJio=

I added a TIMEOUT message when the parser takes more then 1 second to complete.

**Comment by ingydotnet** [comment]
Related, directives must be follow by a `---` line or the parser should error.

See https://play.yaml.io/main/parser?input=JVhZWgo=
vs https://play.yaml.io/main/parser?input=JVhZWgotLS0K

**Comment by aaubry** [comment]
A fix for this issue has been released in [version 12.0.0](https://github.com/aaubry/YamlDotNet/releases/tag/v12.0.0).
